Use even numbers for path ID (#331)
quicwg/multipath: draft-ietf-quic-multipath.md

Path Identifier. When there is not enough unused Path Identifiers, endpoints SHOULD
send MAX_PATHS frame to inform the peer that new Path Identifiers are available.

## Using even numbers for path ID

This specification requires path IDs to be even numbers. We anticipate
that future extensions will lift the requirement in {{path-initiation}} that
paths are only initiated by clients. In that case, to avoid collisions,
we will need separate path IDs number spaces for client-initiated and

> server initiated spaces, such as using even numbered path IDs for client initiated paths

Reviewer: ```suggestion
server-initiated paths, such as using even numbered path IDs for client-initiated paths
```

> and odd numbered path IDs for server initiated paths.

Reviewer: ```suggestion
and odd numbered path IDs for server-initiated paths.
```

A 128-bit value that will be used for a stateless reset when the associated
connection ID is used.

The Path Identifier MUST be an even number. MP_NEW_CONNECTION_ID received as
with an odd numbered path identifier MUST be treated as
a connection error of type MP_PROTOCOL_VIOLATION.

The Path Identifier MUST NOT be larger than the value authorized in the
last MAX_PATHS frame received from the peer, or the value received
from the peer in
the initial_max_paths transport parameters if no MAX_PATHS frames have

> been received. MP_NEW_CONNECTION_ID received as

Reviewer: ```suggestion
been received. MP_NEW_CONNECTION_ID received
```

> with a laue larger than authorized MUST be treated as

Reviewer: ```suggestion
with Path ID value larger than authorized MUST be treated as
```

MAX_PATHS frames contain the following field:

Maximum Paths:
: A count of the cumulative number of path that can be opened
over the lifetime of the connection.

The endpoint receiving this frame MAY send MP_NEW_CONNECTION_ID frames
with Path IDs strictly lower than twice the Maximum Paths.

> For example, if an endpoint received a Maximum Paths value of 2,
> it may send MP_NEW_CONNECTION_ID with Path ID 0 and 2, but not higher.

Reviewer 2: A small suggestion to make this clearer (and avoid the potential reader ambiguity of the "2" value)
```suggestion
For example, if an endpoint received a Maximum Paths value of 3,
it may send MP_NEW_CONNECTION_ID with Path ID 0, 2 and 4, but not higher.
```

Author: Yes, good idea.
